flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -98,4 +102,24 @@ public static TableSchema toSchema(RowType rowType) {
     }
     return builder.build();
   }
+
+  /**
+   * Prune columns from a {@link Schema} using a projected fields.
+   * TODO Why Spark care about filters?
```

Comment:
The columns which has been involved in push-down filter must be in the projection column list. Because just like spark:
`Spark doesn't support residuals per task, so return all filters to get Spark to handle record-level filtering`. Flink source also doesn't support residuals per task, left these filtering to Flink planner.